src/test/java/org/folio/rest/impl/FiscalYearTest.java

```diff
@@ -0,0 +1,111 @@
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import java.util.List;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.jaxrs.model.FiscalYear;
+import org.folio.rest.util.ErrorCodes;
+import org.folio.rest.util.MockServer;
+import org.junit.jupiter.api.Test;
+
+public class FiscalYearTest extends ApiTestBase {
+  private static final Logger logger = LoggerFactory.getLogger(FiscalYearTest.class);
+
+  @Test
+  public void testPostFiscalYearWithCurrency() {
+    logger.info("=== Test create FiscalYear with Currency record populated===");
+
+    Response response = verifyPostResponse(FISCAL_YEAR.getEndpoint(), FISCAL_YEAR.getMockObject(), APPLICATION_JSON,
+        CREATED.getStatusCode());
+    assertThat(response.getBody()
+      .as(FiscalYear.class)
+      .getCurrency(), notNullValue());
+  }
+
+  @Test
+  public void testPostFiscalYearNoConfigurationEntry() {
+    logger.info("=== Test create FiscalYear with default currency record ===");
+    Headers headers = prepareHeaders(X_OKAPI_URL, EMPTY_CONFIG_X_OKAPI_TENANT, X_OKAPI_TOKEN);
+    Response response = verifyPostResponse(FISCAL_YEAR.getEndpoint(), FISCAL_YEAR.getMockObject(), headers, APPLICATION_JSON,
+        CREATED.getStatusCode());
+    assertThat(response.getBody()
+      .as(FiscalYear.class)
+      .getCurrency(), notNullValue());
+  }
+
+  @Test
+  public void testPostFiscalYearErrorConfigurationEntry() {
+    logger.info("=== Test create FiscalYear with Error ===");
+    Headers headers = prepareHeaders(X_OKAPI_URL, ERROR_X_OKAPI_TENANT, X_OKAPI_TOKEN);
+    verifyPostResponse(FISCAL_YEAR.getEndpoint(), FISCAL_YEAR.getMockObject(), headers, APPLICATION_JSON,
+        INTERNAL_SERVER_ERROR.getStatusCode()).as(Errors.class);
+  }
+
+  @Test
+  public void testPutFiscalYearWithCurrency() {
+    logger.info("=== Test update FiscalYear with Currency record populated===");
+
+    JsonObject body = FISCAL_YEAR.getMockObject();
+    verifyPut(FISCAL_YEAR.getEndpointWithId((String) body.remove(ID)), body, "", NO_CONTENT.getStatusCode());
+
+    List<JsonObject> rqRsPostFund = MockServer.getRqRsEntries(HttpMethod.PUT, FISCAL_YEAR.name());
+    assertThat(rqRsPostFund.get(0)
+      .getString("currency"), notNullValue());
+  }
+
+  @Test
+  public void testPutFiscalYearNoConfigurationEntry() {
+    logger.info("=== Test update FiscalYear with default currency record ===");
+
+    Headers headers = prepareHeaders(X_OKAPI_URL, EMPTY_CONFIG_X_OKAPI_TENANT, X_OKAPI_TOKEN);
+    JsonObject body = FISCAL_YEAR.getMockObject();
+
+    verifyPut(FISCAL_YEAR.getEndpointWithId((String) body.remove(ID)), body.toString(), headers, "", NO_CONTENT.getStatusCode());
+
+    List<JsonObject> rqRsPostFund = MockServer.getRqRsEntries(HttpMethod.PUT, FISCAL_YEAR.name());
+    assertThat(rqRsPostFund.get(0)
+      .getString("currency"), notNullValue());
+    assertEquals(rqRsPostFund.get(0)
```

Comment:
Sonar cloud detected code smell. Seems like the actual and expected arguments should be swapped.